compiler/erlang_check.erl

```diff
@@ -279,6 +275,113 @@ check_module(File) ->
     end.
 
 %%------------------------------------------------------------------------------
+%% @doc Check for some known files and try to guess what build system is being
+%% used.
+%% @end
+%%------------------------------------------------------------------------------
+-spec guess_build_system(string()) -> {atom(), string()}.
+guess_build_system(Path) ->
+    BuildSystems = [
+                    {rebar, [
+                             "rebar.config",
+                             "rebar.config.script"
+                            ]
+                    },
+                    {make, [
+                            "Makefile"
+                           ]
+                    }
+                   ],
+    case guess_build_system(Path, BuildSystems) of
+        {_, []} -> unknown_build_system;
+        {BuildSystem, Files} -> {BuildSystem, Files}
+    end.
+
+guess_build_system(Path, [{BuildSystem, Files}|Rest]) ->
+    log("Try build system: ~p ~p~n", [BuildSystem]),
+    case find_files(Path, Files) of
+        [] -> guess_build_system(Path, Rest);
+        FoundFiles when is_list(FoundFiles) -> {BuildSystem, FoundFiles}
+    end.
+
+find_files(Path, Files) ->
+    find_files(Path, Files, Files).
+
+find_files("/", _Files, _AllFiles) ->
+    [];
+find_files(_Path, [], _AllFiles) ->
+    [];
+find_files(Path, [File|Rest], AllFiles) ->
+    AbsFile = absname(Path, File),
+    case filelib:is_regular(AbsFile) of
+        true -> log("Found build file: ~p~n", [AbsFile]), [AbsFile];
+        false -> [] ++ find_files(Path, Rest, AllFiles)
+    end ++ find_files(filename:dirname(Path), AllFiles, AllFiles).
```

Comment:
The double recursion took me some time to understand... I suggest using two simple recursive functions instead of a complex one:

```erlang
%%------------------------------------------------------------------------------
%% @doc For each directory between Dir and "/", collect the first file that has
%% one of the names specified in Files.
%% @end
%%------------------------------------------------------------------------------
-spec find_files(string(), [string()]) -> [string()].
find_files("/", Files) ->
    find_file("/", Files);
find_files(Dir, Files) ->
    find_file(Dir, Files) ++
    find_files(filename:dirname(Dir), Files).

%%------------------------------------------------------------------------------
%% @doc Find the first filename present in a directory.
%%
%% This function returns either an empty list (if there is no such file) or a
%% list containing one element.
%% @end
%%------------------------------------------------------------------------------
-spec find_file(string(), [string()]) -> [string()].
find_file(_Dir, []) ->
    [];
find_file(Dir, [File|Rest]) ->
    AbsFile = absname(Dir, File),
    case filelib:is_regular(AbsFile) of
        true ->
            log("Found build file: ~p~n", [AbsFile]),
            [AbsFile];
        false ->
            find_file(Dir, Rest)
    end.
```

This also finds files directly in the root directory. (Not a common scenario I guess, but in the world of virtualization and Docker and chroot, one never knows.)